Read NodeJS-versions from nodejs.org/dist/index.json

Myhlamaeus opened the pull request:
This is almost the same as #30, with the exception of a different data source.

* Add an option `removeOld`, which removes all unsupported versions from the current `.travis.yml` (ie it just ignores `node_js` in it).
* Read the list of NodeJS-versions from <https://nodejs.org/dist/index.json> and add all LTS-releases and the most recent release to the `.travis.yml`
  * This will automatically support all new releases (unless the format is changed)
  * I don’t know how the EOL of the LTS-releases will be reflected in the data. It would work without further adjustments if `lts` would just be set to `false`.
  * The EOL-dates can’t be added to the `.travis.yml`, as the data don’t contain them.
  * An option to use an HTTP proxy should be added at some point (related issue: yeoman/generator-node#274)
  
EDIT: I’ve just noticed that the HTTP proxy thing in this instance is entirely unrelated to `github-username`.

Myhlamaeus committed:
feat: use the newest and all LTS-versions

Read all LTS-versions and the newest version from
https://nodejs.org/dist/index.json.

Myhlamaeus committed:
feat: add option `removeOld`

Add the option `removeOld`: ‘Remove old versions from the current .travis.yml.’

Myhlamaeus committed:
chore: update package-lock.json

SBoudrias commented on generators/app/index.js:
```diff
@@ -5,6 +5,25 @@ var sort = require('sort-object');
 var ramda = require('ramda');
+var got = require('got');
+
+var supportedVersions = got('https://nodejs.org/dist/index.json', {
```
We should handle the failure case too; if it fails, we just ignore and defaults to a simple subset? Like only `- node`

Myhlamaeus replied:
Wouldn’t `- lts/*` make more sense? We could also guess the currently supported versions through the release schedule.

SBoudrias replied:
Yeah, I just think this is an edge case, so we should do the simplest thing with lowest maintenance cost

Myhlamaeus replied:
I’ve tried this locally (by disabling the internet connection) and it’s working, but I’m not really sure on how to add tests for this.

SBoudrias commented on generators/app/index.js:
```diff
@@ -33,13 +59,24 @@ module.exports = class extends Generator {
           )
         )
       : {};
+    if (this.options.removeOld) {
```
I'd prefer we don't add a new option. I think it makes sense to just always remove the old version; and it'd easily fit the 95% use case.

Myhlamaeus committed:
Revert "feat: add option `removeOld`"

This reverts commit c96ffd7ac86adbaece6c5304004ae5c2ff9ef29d.

Myhlamaeus committed:
feat: discard existing `node_js`-versions

BREAKTING CHANGE: The `node_js`-field is ignored in the existing
`.travis.yml`. Use the `config`-option to supply specific values.

Myhlamaeus committed:
fixup! feat: use the newest and all LTS-versions

Myhlamaeus committed:
fixup! feat: discard existing `node_js`-versions

Myhlamaeus committed:
fixup! fixup! feat: use the newest and all LTS-versions

SBoudrias commented:
Thanks for the awesome PR!!
